components/automate-cli/cmd/chef-automate/ha_node_delete_aws.go

```diff
@@ -101,40 +89,54 @@ func (dna *DeleteNodeAWSImpl) Execute(c *cobra.Command, args []string) error {
 			return nil
 		}
 	}
-	dna.prepare()
 
-	return dna.runDeploy()
-}
-func (dna *DeleteNodeAWSImpl) modifyConfig() error {
-	dna.config.Architecture.ConfigInitials.Architecture = "aws"
-	err := modifyConfigForDeleteNodeForAWS(
-		&dna.config.Automate.Config.InstanceCount,
-		dna.automateIpList,
-	)
+	err = dna.prepare()
 	if err != nil {
-		return status.Wrap(err, status.ConfigError, "Error modifying automate instance count")
+		return err
 	}
-	err = modifyConfigForDeleteNodeForAWS(
-		&dna.config.ChefServer.Config.InstanceCount,
-		dna.chefServerIpList,
-	)
+
+	err = dna.runDeploy()
 	if err != nil {
-		return status.Wrap(err, status.ConfigError, "Error modifying chef-server instance count")
+		return err
 	}
-	err = modifyConfigForDeleteNodeForAWS(
-		&dna.config.Opensearch.Config.InstanceCount,
-		dna.opensearchIpList,
-	)
-	if err != nil {
-		return status.Wrap(err, status.ConfigError, "Error modifying opensearch instance count")
+
+	return dna.stopNodes()
```

Comment:
do this inside a check if instance count has decreased or not. You can get instance count using terraform state list command

```
err = dna.runDeploy()
if err != nil {
		dna.writer.Println(err)
}
if ExpectedInstanceCount() = CurrentInstanceCount{
        return dna.stopNodes()
}
```

Reply:
done